app/models/course/group.rb

```diff
@@ -28,7 +29,7 @@ def set_defaults
   def should_create_manager?
     course && creator &&
       course.course_users.exists?(user: creator) &&
-      !group_users.exists?(user: creator)
+      !group_users.any? { |group_user| group_user.course_user.user_id == creator }
```

Comment:
Would the solution be `!group_users.course_users.includes(:user).any? { |group_user| group_user.course_user.user_id == creator }`?